src/components/correct-name/CorrectCompanyName.vue

```diff
@@ -36,8 +36,13 @@ export default class CorrectCompanyName extends Vue {
   @Prop({ required: true }) readonly validate!: boolean
 
   // Local properties
-  formValid = false // initially invalid
-  textfield = this.companyName
+  @Provide() formValid = false // initially invalid
+  @Provide() textfield = ''
+
+  mounted (): void {
+    // Filter the options to be displayed by what id's were passed from the parent component
+    this.textfield = this.companyName
+  }
```

Comment:
This assignment only happens once, when this component is mounted. Can the companyName prop change after the component is mounted? If so then you should watch the prop instead, eg,
```
@Watch('companyName', { immediate: true })
onCompanyNameChanged (val: string) {
  this.textfield = val
}
```

Reply:
I think it is much better.